test/cpp/tensorexpr/test_simplify.cpp

```diff
@@ -1235,6 +1235,118 @@ TEST(Simplify, SimplifyDiv) {
   }
 }
 
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+TEST(Simplify, SimplifyDivWithLoopContext1) {
+  KernelScope kernel_scope;
+  // Stmt to simplify:
+  // for (int i = 0; i < 5; i++) {
+  //  A[i] = (i + 24) / 6;
+  //}
+  VarHandle i("i", kInt);
+  BufHandle a_buf("A", {5}, kInt);
+  auto forstmt = For::make(
+      i, 0, 5, Store::make(a_buf, {i}, (i + ExprHandle(24)) / ExprHandle(6)));
+
+  const Stmt* simplified = IRSimplifier::simplify(forstmt);
+
+  std::ostringstream oss;
+  oss << *(simplified);
+  const std::string& verification_pattern =
+      R"IR(
+# CHECK: for (int i
+# CHECK-NEXT:   A[i] = 4;
+      )IR";
+  torch::jit::testing::FileCheck().run(verification_pattern, oss.str());
+}
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+TEST(Simplify, SimplifyDivWithLoopContext2) {
+  KernelScope kernel_scope;
+  // Stmt to simplify:
+  // for (int i = 0; i < 5; i++) {
+  //  for (int j = 0; j < 5; j++) {
+  //    A[i, j] = (i + 6*j) / 6;
```

Comment:
The optimization should be valid if j iterates beyond 6, right? Could you please add a test for this?

Reply:
This comment is still relevant.

Reply:
Yeah, it should be valid if j is greater than 6. Will add a test!